### DISCUSSION TITLE
Missing properties whose specification is required in certain scenarios from XRechnung 3.0 onwards

### DISCUSSION THREAD

**Discussion by ErikTr** [stephanstapel/ZUGFeRD-csharp#227] [General]
The version history of the current [XRechnung 3.0.1 standard](https://xeinkauf.de/app/uploads/2023/09/301-XRechnung-2023-09-22.pdf), as shown in Appendix C, specifies certain properties as mandatory in certain scenarios. However, some of these properties are not supported by ZUGfERD-csharp. For instance, the TradeAllowanceCharge class lacks a property for [percentage allowance/charge](https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-AllowanceCharge/cbc-MultiplierFactorNumeric/). Will the partially mandatory properties resulting from the harmonization of XRechnung with the Peppol BIS Billing 3.0 standard be added before XRechnung 3.0 comes into force?

**Comment by stephanstapel** [comment]
Hi Erik,
This looks like a quite big change that I probably will not implement soon.
I am happy to review pull requests. Alternatively I am happy to discuss offline about a consulting task.
